### PR TITLE
Add DldPulses interface for pulse information in reconstructed DLD data

### PR DESCRIPTION
**Pull request by philsmt**
First `PulsePattern` implementation based on the new and simpler interface.

The offline calibration based REMI/DLD reconstruction adds pulse information to the output files that *may* be different than the one obtained from timeserver data due to particular configurations chosen for reconstruction at the time. This compoent allows to read this data and expose it via the very same interface.

### CODE REVIEW

**Review by JamesWrigley — CHANGES_REQUESTED** [review]
Minor nitpicking :mag: This should also be documented in `components.md`.

File: src/extra/components/pulses.py [hunk 1]
```diff
@@ -778,3 +778,84 @@ def _get_ppdecoder_node(self):
...
+        import pandas as pd
+        index = pd.MultiIndex.from_arrays(
+            list(index_levels.values()), names=list(index_levels.keys()))
+
+        if 'pulse' in triggers.dtype.fields:
+            pulse_ids = triggers['pulse'].copy()
+        else:
+            # Try to guess pulse IDs from trigger positions.
+            import sys
+            print(f'{self.__class__.__name__}._get_pulse_ids(): No actual '
```
[the remark this answers is not in the document]
Comment (philsmt) [reply src/extra/components/pulses.py:837]: It does, but it runs into the same problem we had in EXtra-data with automatic aliases - it's hidden by default. I'm fine with removing if you feel strongly about it, but I see no other way to *really* make it visible for those that may trip over it during an analysis.

File: docs/changelog.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ Added:
 
 Changed:
 
+- [DldPulses][extra.components.DldPulses] to access pulse information saved during delay line detector event reconstruction (!42).
```
[the remark this answers is not in the document]
Comment (philsmt) [reply docs/changelog.md:26]: Oops!

File: src/extra/components/pulses.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesWrigley) [reply src/extra/components/pulses.py:840]: Ah right I forgot about that, NVM then.

**Review by JamesWrigley — APPROVED** [review]
LGTM!

**Comment by philsmt** [comment]
Thank you!

**Pull request merged**